moto/glue/models.py

```diff
@@ -183,6 +183,32 @@ def create_job(
         )
         return name
 
+    def list_jobs(self, next_token, max_results, tags):
+        next_token = int(next_token) if next_token else 0
+        total_number_of_jobs = len(self.jobs)
+        max_results = max_results or total_number_of_jobs
+        last_token = next_token + max_results
+        job_names = self.filter_jobs_by_tags(tags, next_token, last_token)
+        return (
+            job_names,
+            last_token if last_token <= total_number_of_jobs else total_number_of_jobs,
```

Comment:
AWS does not return a value for `next_token` if there are no more results (or rather: `next_token=None`)

If a developer wants to retrieve all jobs, they would create a loop like this:
```
while next_token:
    jobs, next_token = list_jobs(next_token)
```
The current implementation would create a never-ending loop, if we always return a value for `next_token`

Reply:
Added `test_next_token_logic_does_not_create_infinite_loop` just to be on the safe side 😄